### ISSUE TITLE
ess-indent-with-fancy-comments  doesn't respect nil

### ISSUE THREAD

**Issue by mmahmoudian** [emacs-ess/ESS#1175]
Before anything, thanks for all your efforts in bringing nice R experience to Emacs.

**Disclaimer:** I'm n00b in Emacs and ESS, but I have done all my best to make sure I've not done something stupid and waste your time. Using multiple search engines and spending over an hour searching and reading didn't help.

## The problem

I have set `(setq ess-indent-with-fancy-comments nil)` but still the `#` will be indented to column 40.

This approach should work as it is in the documentation (`C-h v ess-indent-with-fancy-comments`) and also is recently mentioned in https://github.com/emacs-ess/ESS/issues/434#issuecomment-992167502 that it is the correct syntax and approach.

The following is my config for ESS:

<details>
<summary>My ESS config</summary>

```elisp
;; ESS specific configuration
(use-package ess
  ;;:ensure t
  :defer t

  :init
  (require 'ess-site)
  ;; (require 'ess-view)
  ;; (require 'ess-R-data-view)
  ;; (require 'poly-R)

  :config
  ;; C-h v  and then ess-style-alist for more information
  (setq ess-style 'OWN)
  (setq ess-indent-with-fancy-comments nil)

  ;; set the indentation and alignment
  ;;   https://ess.r-project.org/Manual/ess.html#index-C_002dj

  ;; ;; auto-width
  ;; (setq ess-auto-width 'window)

  ;; ;; let lsp manage lintr
  ;; (setq ess-use-flymake nil)

  ;; Stop R repl eval from blocking emacs.
  (setq ess-eval-visibly 'nowait)

  ;; (setq ess-ask-for-ess-directory t
  ;;       ess-local-process-name "R"
  ;;       ansi-color-for-comint-mode 'filter
  ;;       comint-scroll-to-bottom-on-input t
  ;;       comint-scroll-to-bottom-on-output t
  ;;       comint-move-point-for-output t)

  ;; Syntax highlighting is nice, so let’s turn all of that on
  (setq ess-R-font-lock-keywords
        '((ess-R-fl-keyword:keywords . t)
          (ess-R-fl-keyword:constants . t)
          (ess-R-fl-keyword:modifiers . t)
          (ess-R-fl-keyword:fun-defs . t)
          (ess-R-fl-keyword:assign-ops . t)
          (ess-R-fl-keyword:%op% . t)
          (ess-fl-keyword:fun-calls . t)
          (ess-fl-keyword:numbers . t)
          (ess-fl-keyword:operators . t)
          (ess-fl-keyword:delimiters . t)
          (ess-fl-keyword:= . t)
          (ess-R-fl-keyword:F&T . t)))

  ;; insert pipes etc...
  (defun my-ess-insert-assign ()
    "Insert an assignment <-"
    (interactive)
    ;; check if we need to insert space
    (if (not (string= (string (preceding-char)) " "))
        (insert " "))
    (insert "<- "))

  (defun my-ess-insert-pipe ()
    "Insert a |> and newline"
    (interactive)
    ;; check if we need to insert space
    (if (not (string= (string (preceding-char)) " "))
        (insert " "))
    (insert "|> "))
  ;; set keybindings
  ;; insert pipe
  (define-key ess-r-mode-map (kbd "M--") 'my-ess-insert-assign)
  (define-key inferior-ess-r-mode-map (kbd "M--") 'my-ess-insert-assign)
  ;; insert assign
  (define-key ess-r-mode-map (kbd "M-<") 'my-ess-insert-pipe)
  (define-key inferior-ess-r-mode-map (kbd "M-<") 'my-ess-insert-pipe)
  )

```

</details>

P.s: I also realized that many of the variables related to "style" that are mentioned in https://ess.r-project.org/Manual/ess.html are deprecated which makes it very hard for a new user to even setup a minimal config like mine. I wonder if there is an automated way to  check if the variables there are still valid or not.

## Systems information

- ess-version: 18.10.3snapshot [<unknown>] (loaded from /home/mehrad/.emacs.d/.local/straight/build-27.2/ess/)
- GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, cairo version 1.17.4) of 2021-03-26
- Doom emacs	v21.12.0-alpha   HEAD -> develop b51d7d6d7 2022-01-05 04:06:28 +0100

As far as I know everything on my computer is using the latest versions. Please let me know if you need more information.

**Comment by vspinu** [comment]
ESS style setup is a bit convoluted for historical reasons. The global style variables take effect only if `ess-style` is `DEFAULT` see the docs for `ess-style-alist`. So your global setting does not take effect. You have 4 options:

 1. Set variables in `ess-mode-hook` and `inferior-ess-mode-hook` (aka define your own `my/ess-options` functions with all your settings and add to the hook)
 2. Set the entire `ess-own-style-list` (or some of those entries) and set `ess-style` to `OWN`, or
 3. Add your own list with style variables into `ess-style-alist` and set `ess-style` to whatever name. 
 4. For your case if the only thing you want is no-fancy comments then you can pick one  of the precefined styles which set it to nil (`Studio-` or `Studio`). In other words the global `(setq ess-style 'Rstudio)` should do the trick. 

You can also change styles on the fly with `ess-set-style` bound by default to `C-c C-e C-s`.

**Comment by mmahmoudian** [comment]
Thanks for the response. I initially used `(setq ess-style 'Rstudio)` but having `(ess-indent-offset . 2)` instead of 4 that I quickly moved to the first thing I saw with `(ess-indent-offset . 4)`.

On line [1321](https://github.com/emacs-ess/ESS/blob/master/lisp/ess-custom.el#L1321) of the `ess.custom.el` it is mentioned that:

> OWN style is defined in `ess-own-style-list' and you can customize it to your needs.

and that was I was trying to do with the config I posted above. Is there a way to set the `ess-indent-with-fancy-comments` in `OWN` to `nil` on the fly before the `(setq ess-style 'OWN)`? considering that the `ess-style-alist` is a variable, I assume it is possible, but I don't know how.
